### PR TITLE
Addition of Pentade on Use Date Dlg

### PR DESCRIPTION
**Pull request by stevenndungu**

### CODE REVIEW

**Comment by stevenndungu** [comment]
@africanmathsinitiative/instat , this is ready for review.

**Comment by stevenndungu** [comment]
Yes your right, for both pentad and dekad. There is a small problem to be
fixed. But otherwise it is a good start I guess.

**Comment by lilyclements** [comment]
My understanding of Pentad was that it would denote
'0' for days 1-5,
'1' for days 6-10,
...
'5' for days 26-End
It does this on the first month. However, on the second month it goes

'3' for days 1-5
'4' for 6-10
....
'8' for 26-End

In the third month:
'6' for days 1-5
'7' for days 6-10
...
'11' for days 26-End.

Have I misunderstood what we expect from pentad or is this an error in the code?

**Comment by stevenndungu** [comment]
Thanks for noting that... I saw that problem as well on dekad.

I noticed that the fault begins when transiting from one month to the other
in both cases( of dekad and pentad).
Still thinking about it,  ie, on how to adjust the method to work
correctly.

If you have ideas on how to improve it- most welcome. Thanks.

**Comment by rdstern** [comment]
There should be 6 pentades in each month, so 12 * 6 = 72 in the year, just as there are 3 dekads in each month, so they should go from 1 to 36.

So, in January level 1 is 1-5, etc, and 26-31 is pentade 6.
Then in February 1-5 is level 7 to 26-28 or 29 is level 12, etc.

So, if numeric they should go from level 1 to level 72.

They could be labelled as Jan1,...Jan6, then Feb1,...Feb6.  Similarly the dekads could be labelled Jan1-Jan3, etc.

Don't bother with full month names.

Hope this helps

Roger

**Comment by stevenndungu** [comment]
Quite helpful, because this implies that the last decade or pentad in each month must necessarily have 50 or 10 days respectively. Which was one thing I did not consider from my interpretation.  Considering months independently of each other. Thanks

**Comment by rdstern** [comment]
I am editing your comment, because I don't understand it.  Is it

Quite helpful, because this implies that the last dekad or pentad in each month must **_not_** necessarily have **_5_** or 10 days respectively. Which was one thing I did not consider from my interpretation. Considering months independently of each other. Thanks

**Comment by lilyclements** [comment]
Great, so dekad is working fine then I just misunderstood what we wish to have for it. It's just pentad which is having the issue. I'm having a look for code and have this

pentad = ceiling( (DOY_366 - leap_year(Year)*(DOY_366 > 59)) / 5 )

However, this does have an issue with the leap years. So, for my data which uses doy_366 for the day:

image

We can see firstly how it affects a non leap year and then a leap year. In the leap year it is not correct as it has gone over by a day

**Comment by stevenndungu** [comment]
@rdstern  I intended to mean that, sorry for slight error.

File: instat/static/InstatObject/R/stand_alone_functions.R [hunk 1]
```diff
@@ -227,6 +227,11 @@ dekade <- function(date) {
   return(temp_dekade)
   }
 
+  pentad <- function(date) {
+  temp_pentad <- 3 * (month(date) - 1) + (mday(date) > 5) + (mday(date) > 10) + (mday(date) > 15) + (mday(date) > 20) + (mday(date) > 25)
```
Comment (lilyclements) [inline instat/static/InstatObject/R/stand_alone_functions.R]: If we do '6*' instead of '3*' then it works for leap years and non-leap years

Comment (stevenndungu) [inline instat/static/InstatObject/R/stand_alone_functions.R]: Great thanks, let me make the changes and see how it works.

**Comment by stevenndungu** [comment]
Great! Working perfectly well.

**Comment by dannyparsons** [comment]
This looks good now. Well done on the writing and testing.

**Pull request merged**